src/results.ts

```diff
@@ -0,0 +1,68 @@
+export interface ResultValues {
+  [id: string]: string | string[];
+}
+
+export interface PosNegToken {
+  pos: string;
+  neg: string;
+}
+
+export interface ClickToken {
+  click: string;
+}
+
+export type Token = PosNegToken | ClickToken;
+
+export interface Result {
+  // Values are field values of records.
+  values: ResultValues;
+
+  // Token contains the token associated with this Result (if any).
+  token: Token;
+
+  // Score is the overall score of this Result.
+  score: number;
+
+  // IndexScore is the index-matched score of this Result.
+  indexScore: number;
+}
+
+// CountResponse is a type returned from a query which has
+// performed a count aggregate.
+export interface CountResponse {
+  [id: string]: number;
+}
+
+export interface BucketResponse {
```

Comment:
if we aren't going to impl this should we remove the types.

Reply:
ACK, types and impl added.